### PR TITLE
enable feature/coupled-crow to work on stampede2

### PR DESCRIPTION
**Pull request by jkuang-exploro**
In addition to the stampede2 porting, this branch also switch to hpc-stack on Orion. Please try run this branch on both Orion and stampede2.

### CODE REVIEW

File: modulefiles/module_base.orion_bk
Comment (JessicaMeixner-NOAA) [inline modulefiles/module_base.orion_bk]: This file is not needed, is there an example test case on orion since the orion modules were updated?

File: sorc/link_fv3gfs.sh
Comment (JessicaMeixner-NOAA) [inline sorc/link_fv3gfs.sh]: These fix files should be pointing to a. fix_UFSp6

File: sorc/machine-setup.sh_bk
Comment (JessicaMeixner-NOAA) [inline sorc/machine-setup.sh_bk]: Is this file needed?

File: workflow/cases/coupled_free_forecast_wave_test.yaml
Comment (JessicaMeixner-NOAA) [inline workflow/cases/coupled_free_forecast_wave_test.yaml]: What is this case file for? Can an existing one not be used?

**Review by JessicaMeixner-NOAA — COMMENTED** [review]
Is there an example directory for output with this change for orion?  I'm guessing someone else will need to test for the UPP changes on hera as well.  Do I need to do this or is someone else available to test this on hera? 

I think the fix files need to be updated on Stampede.

File: modulefiles/module_base.orion_bk
Comment (jkuang-exploro) [reply modulefiles/module_base.orion_bk]: The same case file for the trunk works here.

File: workflow/cases/coupled_free_forecast_wave_test.yaml
Comment (jkuang-exploro) [reply workflow/cases/coupled_free_forecast_wave_test.yaml]: This file is excessive, along with the _bk file mentioned above. I will remove them.

File: sorc/link_fv3gfs.sh
Comment (jkuang-exploro) [reply sorc/link_fv3gfs.sh]: I will update it.

**Comment by jkuang-exploro** [comment]
@JessicaMeixner-NOAA Thanks for these comments. I think since the primary purpose of this PR is to enable stampede2 running (stampede2 ONLY has hpc-stack modules, as of I know), and coupled-crow is functional on Hera as of now, we should leave hera-hpcstack adoption to a future PR.

**Comment by JessicaMeixner-NOAA** [comment]
> @JessicaMeixner-NOAA Thanks for these comments. I think since the primary purpose of this PR is to enable stampede2 running (stampede2 ONLY has hpc-stack modules, as of I know), and coupled-crow is functional on Hera as of now, we should leave hera-hpcstack adoption to a future PR.

Agreed, however I think hera still needs to be tested for this PR because the version of ncep_post and how it was built is changed.  Therefore, I think we need to ensure nothing was broken on hera.  In addition, there are updates for orion in this PR, so we should make sure it's been tested there as well, even though that was not the main goal of this PR, which as you stated is the enable stampede2.

**Comment by jkuang-exploro** [comment]
@JessicaMeixner-NOAA Yes. I agree.

**Comment by JessicaMeixner-NOAA** [comment]
@JianKuang-NOAA  I noticed this branch is out of date when I was checking it out to test on hera, can you please merge the latest feature/coupled-crow so I can test this on hera -- if someone else is already running the test on hera, please let me know.

**Comment by jkuang-exploro** [comment]
@JessicaMeixner-NOAA Thanks for your mention. I just merged it.

File: ush/load_fv3gfs_modules.sh
Comment (JessicaMeixner-NOAA) [inline ush/load_fv3gfs_modules.sh]: do we want this uncommented or was this just for debugging?

File: modulefiles/module_base.orion_bk
Comment (JessicaMeixner-NOAA) [reply modulefiles/module_base.orion_bk]: Should this be removed?

**Comment by JessicaMeixner-NOAA** [comment]
@JianKuang-NOAA my attempt on hera failed, but the error seemed weird so I'm trying again with a fresh clone/build and will let you know how that goes.  In the meantime my test on orion failed for ocean and the gridded wave post: /work/noaa/marine/jmeixner/p6stampede/test01/COMROOT/test01/logs/2013040100 
It's my understanding that those currently work on orion, so hopefully those can be fixed before merging this PR in.

**Review by JessicaMeixner-NOAA — CHANGES_REQUESTED** [review]
The re-test on hera went fine and @LydiaStefanova-NOAA checked the atm-post and the grib files were identical to what is in the current feature/coupled-crow so that's all fine.  The orion issues (wave post and ocean post not running) need to be addressed before merging.

**Comment by WalterKolczynski-NOAA** [comment]
@JianKuang-NOAA What's the status of this?

**Comment by jkuang-exploro** [comment]
@WalterKolczynski-NOAA I am waiting for the update for ocean post enabling it to run using hpc-stack.

**Comment by WalterKolczynski-NOAA** [comment]
> @WalterKolczynski-NOAA I am waiting for the update for ocean post enabling it to run using hpc-stack.

reg2grb2 or something else? Because the PR migrating reg2grb2 to hpc-stack was merged last week (#273).

**Comment by jkuang-exploro** [comment]
@WalterKolczynski-NOAA Yes. That's great. I will merge-in the updated coupled-crow and re-enable this PR later today. And invite you to review.

File: modulefiles/module_base.orion_bk
Comment (WalterKolczynski-NOAA) [reply modulefiles/module_base.orion_bk]: Agree this should be removed

File: modulefiles/module_base.stampede
Comment (WalterKolczynski-NOAA) [inline modulefiles/module_base.stampede:7]: We need to talk to the stack team about getting 1.1.0 on stampede to match the other systems, so the other module versions can match as well.

File: modulefiles/modulefile.reg2grb2.stampede
Comment (WalterKolczynski-NOAA) [inline modulefiles/modulefile.reg2grb2.stampede]: Comment has wrong machine.

File: modulefiles/modulefile.ww3.stampede
Comment (WalterKolczynski-NOAA) [inline modulefiles/modulefile.ww3.stampede]: Are these available/necessary on stampede? I don't see them.

File: sorc/build_ncep_post.sh
Comment (WalterKolczynski-NOAA) [inline sorc/build_ncep_post.sh]: The submodule update should be done in the checkout script.

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
Minor changes except for getting hpc-stack updated.

File: modulefiles/module_base.stampede
Comment (WalterKolczynski-NOAA) [reply modulefiles/module_base.stampede:7]: Just talked to @kgerheiser, he's going to get 1.1.0 on Stampede.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
Approved once the component updates are in.

**Comment by jkuang-exploro** [comment]
@WalterKolczynski-NOAA This PR is now up-to-date. Please take a test on Hera and Orion.

File: sorc/link_fv3gfs.sh
Comment (JessicaMeixner-NOAA) [inline sorc/link_fv3gfs.sh]: Why is this commented out?

File: modulefiles/module_base.orion_bk
Comment (JessicaMeixner-NOAA) [reply modulefiles/module_base.orion_bk]: This file needs to be removed before PR is merged.

**Review by JessicaMeixner-NOAA — CHANGES_REQUESTED** [review]

File: sorc/link_fv3gfs.sh
Comment (jkuang-exploro) [reply sorc/link_fv3gfs.sh]: This is for debug purpose. I am restoring it now.

**Comment by jkuang-exploro** [comment]
@JessicaMeixner-NOAA @WalterKolczynski-NOAA I addressed Jessica's comments. Please take a test and approve this PR if ready. Thanks.

**Comment by WalterKolczynski-NOAA** [comment]
I'm on leave until Monday, so I might not get to it until then.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Comment by WalterKolczynski-NOAA** [comment]
@JessicaMeixner-NOAA Are all your concerns addressed? I'd like to get this in.

**Review by JessicaMeixner-NOAA — APPROVED** [review]
@WalterKolczynski-NOAA yes, my concerns were corrected.  Thanks @JianKuang-NOAA for fixing those. 

I do have two questions though: 1. I thought this was waiting for the latest commit of ufs-weather-model which ported things to stampede: https://github.com/ufs-community/ufs-weather-model/commit/37f0da7b82e32a15bf072eaa158399a6b0f2722c and this PR does not update what is checked out to point to that.  2. the modules used in that PR are for hpc stack 1.1.0 which is inconsistent with this.  

That being said I have no objection to this going in as it doesn't effect hera/orion/etc.

**Comment by WalterKolczynski-NOAA** [comment]
Agree, both of those should be addressed. Then this PR can go in.

**Comment by jkuang-exploro** [comment]
@WalterKolczynski-NOAA Could you make these changes? I am focusing on EWOK project now, plus I have no Hera & WCOSS access which means someone other than me need to test it there anyway.

**Comment by WalterKolczynski-NOAA** [comment]
Since I have to take over, I'm going to merge this PR (since it doesn't break anything), then start from my own clone to update the remaining bits. That is easier than making sure I have all the right permissions to update Jian's branch.

**Comment by WalterKolczynski-NOAA** [comment]
Actually, I just hadn't added Stampede's ssh key to GitHub yet. Oh well, already done. There will be a new PR for the cleanup.

**Comment by WalterKolczynski-NOAA** [comment]
This PR should've tagged Issue #280

**Pull request merged**
